src/Features/CSharp/Portable/MakeMethodSynchronous/RemoveAsyncModifierHelpers.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeMethodSynchronous
+{
+    internal static class RemoveAsyncModifierHelpers
+    {
+        internal static SyntaxNode WithoutAsyncModifier(MethodDeclarationSyntax method, TypeSyntax returnType)
+        {
+            var newModifiers = RemoveAsyncModifier(method.Modifiers, ref returnType);
+            return method.WithReturnType(returnType).WithModifiers(newModifiers);
+        }
+
+        internal static SyntaxNode WithoutAsyncModifier(LocalFunctionStatementSyntax localFunction, TypeSyntax returnType)
+        {
+            var newModifiers = RemoveAsyncModifier(localFunction.Modifiers, ref returnType);
+            return localFunction.WithReturnType(returnType).WithModifiers(newModifiers);
+        }
+
+        internal static SyntaxNode WithoutAsyncModifier(ParenthesizedLambdaExpressionSyntax lambda)
+            => lambda.WithAsyncKeyword(default).WithPrependedLeadingTrivia(lambda.AsyncKeyword.LeadingTrivia);
+
+        internal static SyntaxNode WithoutAsyncModifier(SimpleLambdaExpressionSyntax lambda)
+            => lambda.WithAsyncKeyword(default).WithPrependedLeadingTrivia(lambda.AsyncKeyword.LeadingTrivia);
+
+        internal static SyntaxNode WithoutAsyncModifier(AnonymousMethodExpressionSyntax method)
+            => method.WithAsyncKeyword(default).WithPrependedLeadingTrivia(method.AsyncKeyword.LeadingTrivia);
```

Comment:
note: @RikkiGibson is working on static lambdas.  it would be good to update our tests here once that is done to have at least one test ot make sure it still works.